flow360/component/simulation/framework/param_utils.py

```diff
@@ -0,0 +1,113 @@
+"""pre processing and post processing utilities for simulation parameters."""
+
+from typing import Optional
+
+import pydantic as pd
+
+from flow360.component.simulation.framework.base_model import Flow360BaseModel
+from flow360.component.simulation.framework.entity_base import EntityBase, EntityList
+from flow360.component.simulation.framework.entity_registry import EntityRegistry
+from flow360.component.simulation.framework.unique_list import UniqueStringList
+from flow360.component.simulation.primitives import Surface, _VolumeEntityBase
+from flow360.component.simulation.unit_system import LengthType
+from flow360.component.simulation.utils import _model_attribute_unlock
+
+
+class AssetCache(Flow360BaseModel):
+    """
+    Note:
+    1. asset_entity_registry will be replacing/update the metadata-constructed registry of the asset when loading it.
+    """
+
+    asset_entity_registry: EntityRegistry = pd.Field(EntityRegistry(), frozen=True)
```

Comment:
can we just call it `registry`?

Reply:
you mean the asset_entity_registry or the EntityRegistry class?

Reply:
`asset_entity_registry`

Reply:
Done